java/test/jmri/jmrit/roster/RosterSpeedProfileTest.java

```diff
@@ -257,13 +263,44 @@ protected void warnShortLong(String s) {
         // Allow speed step table to be constructed
         //JUnitUtil.waitFor(3000);
         // Note it must be a perfect 0.20
-        //Assert.assertEquals("Speed didnt get to a perfect zero", 0.20f, throttle.getSpeedSetting(), 0.0f);
         JUnitUtil.waitFor(()->(throttle.getSpeedSetting() == 0.20f),"Failed to reach requested speed");
 
+        JUnitAppender.assertWarnMessageStartsWith("distance remaining is now 0, but we have not reached desired speed setting 0.2 v 0.3");
+
         // as the calc goes wrong we immediatly set speed to final speed. The entries are rubbish so dont bother checking
         Assert.assertEquals("SpeedStep Table has lincorrect number of entries.", 1, sp.getSpeedStepTrace().size() ) ;
         sp.cancelSpeedChange();
     }
+
+    @Test
+    public void testconvertThrottleSettingToScaleSpeedWithUnits(){
+
+        SignalSpeedMap ssm = InstanceManager.getDefault(SignalSpeedMap.class);
+        setSpeedInterpretation(ssm, SignalSpeedMap.PERCENT_NORMAL);
+        Assertions.assertEquals("0.50 millimeters/sec",RosterSpeedProfile.convertMMSToScaleSpeedWithUnits(0.5f));
+
+        setSpeedInterpretation(ssm, SignalSpeedMap.PERCENT_THROTTLE);
+        Assertions.assertEquals("0.50 millimeters/sec",RosterSpeedProfile.convertMMSToScaleSpeedWithUnits(0.5f));
+
+        setSpeedInterpretation(ssm, SignalSpeedMap.SPEED_KMPH);
+        Assertions.assertEquals("0.16 Kilometers/Hour",RosterSpeedProfile.convertMMSToScaleSpeedWithUnits(0.5f));
+
+        setSpeedInterpretation(ssm, SignalSpeedMap.SPEED_MPH);
+        Assertions.assertEquals("0.10 Miles/Hour",RosterSpeedProfile.convertMMSToScaleSpeedWithUnits(0.5f));
+
+    }
+
+    private void setSpeedInterpretation(SignalSpeedMap map, int interpretation) {
+        HashMap<String, Float> newMap = new HashMap<>(11);
```

Comment:
I suggest map.size() instead of 11 here in case someone adds a new value to the map.

Reply:
Thanks for looking over @danielb987 , I've tweaked the HashMap generation to avoid magic numbers,
Steve.